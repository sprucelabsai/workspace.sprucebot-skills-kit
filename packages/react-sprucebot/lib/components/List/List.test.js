'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

require('jest-styled-components');

var _List = require('./List');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

test('List renders', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.List, null)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('ListItem should render non-arrays', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(
		_List.ListItem,
		null,
		'YOLO'
	)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('List render send padding-bottom if its piled', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.List, { alignItems: 'flex-end', pile: true })).toJSON();
	expect(tree).toMatchSnapshot();
});

test('ListItem renders', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.ListItem, null)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('ListItem should align items', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.ListItem, { alignItems: 'flex-end' })).toJSON();
	expect(tree).toMatchSnapshot();
});

test('ListItemAvatar should align where we specify', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.ListItem, { alignItems: 'flex-start', avatar: true })).toJSON();
	expect(tree).toMatchSnapshot();
});

test('ItemRightContent should align where we specify', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_List.ListItem, { alignItems: 'flex-start', avatar: true })).toJSON();
	expect(tree).toMatchSnapshot();
});

test('List should render title', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(
		_List.List,
		null,
		_react2.default.createElement(_List.ListItem, { title: 'foo' })
	)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('List should render subTitle', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(
		_List.List,
		null,
		_react2.default.createElement(_List.ListItem, { title: 'foo', subtitle: 'bar' })
	)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('List should render listitems and divs', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(
		_List.List,
		null,
		_react2.default.createElement(
			'div',
			null,
			_react2.default.createElement(_List.ListItem, { title: 'foo', subtitle: 'bar' }),
			_react2.default.createElement(
				'a',
				null,
				'Im also a list item'
			)
		)
	)).toJSON();
	expect(tree).toMatchSnapshot();
});

test('List should render ItemRightContent with a title', function () {
	var tree = _reactTestRenderer2.default.create(_react2.default.createElement(
		_List.List,
		null,
		_react2.default.createElement(
			'div',
			null,
			_react2.default.createElement(_List.ListItem, { rightTitle: 'foo', subtitle: 'bar' }),
			_react2.default.createElement(
				'a',
				null,
				'Im also a list item'
			)
		)
	)).toJSON();
	expect(tree).toMatchSnapshot();
});