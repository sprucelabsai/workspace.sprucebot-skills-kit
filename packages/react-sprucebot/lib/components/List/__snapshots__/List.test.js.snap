// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRightContent should align where we specify 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  margin: 0 10px 0 0;
  border-radius: 50%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemAvatar avatar__outer__wrapper c1"
  >
    <div
      className="avatar__wrapper online "
    />
  </div>
  <div
    className="ItemDetail item__details c2"
  />
</div>
`;

exports[`List render send padding-bottom if its piled 1`] = `
.c0 {
  padding-bottom: 1.25em;
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

<div
  className="List item__list c0"
/>
`;

exports[`List renders 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

<div
  className="List item__list c0"
/>
`;

exports[`List should render ItemRightContent with a title 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c4 {
  text-align: right;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c5 {
  font-weight: 400;
}

.c3 {
  font-size: 0.75em;
}

<div
  className="List item__list c0"
>
  <div>
    <div
      className=" ListItemWrapper item__list__item  c1"
    >
      <div
        className="ItemDetail item__details c2"
      >
        <div
          className="ItemSubTitle sub__title c3"
        >
          bar
        </div>
      </div>
      <div
        className="ItemRightContent content__right c4"
      >
        <div
          className="ItemTitle title c5"
        >
          foo
        </div>
      </div>
    </div>
    <a>
      Im also a list item
    </a>
  </div>
</div>
`;

exports[`List should render listitems and divs 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  font-weight: 500;
}

.c4 {
  font-size: 0.75em;
}

<div
  className="List item__list c0"
>
  <div>
    <div
      className=" ListItemWrapper item__list__item  c1"
      title="foo"
    >
      <div
        className="ItemDetail item__details c2"
      >
        <div
          className="ItemTitle title c3"
        >
          foo
        </div>
        <div
          className="ItemSubTitle sub__title c4"
        >
          bar
        </div>
      </div>
    </div>
    <a>
      Im also a list item
    </a>
  </div>
</div>
`;

exports[`List should render subTitle 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  font-weight: 500;
}

.c4 {
  font-size: 0.75em;
}

<div
  className="List item__list c0"
>
  <div
    className=" ListItemWrapper item__list__item  c1"
    title="foo"
  >
    <div
      className="ItemDetail item__details c2"
    >
      <div
        className="ItemTitle title c3"
      >
        foo
      </div>
      <div
        className="ItemSubTitle sub__title c4"
      >
        bar
      </div>
    </div>
  </div>
</div>
`;

exports[`List should render title 1`] = `
.c0 {
  word-wrap: break-word;
  overflow-wrap: break-word;
  word-break: break-word;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c3 {
  font-weight: 500;
}

<div
  className="List item__list c0"
>
  <div
    className=" ListItemWrapper item__list__item  c1"
    title="foo"
  >
    <div
      className="ItemDetail item__details c2"
    >
      <div
        className="ItemTitle title c3"
      >
        foo
      </div>
    </div>
  </div>
</div>
`;

exports[`ListItem renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  />
</div>
`;

exports[`ListItem should align items 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  />
</div>
`;

exports[`ListItem should render non-arrays 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemDetail item__details c1"
  >
    YOLO
  </div>
</div>
`;

exports[`ListItemAvatar should align where we specify 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  margin: 0 10px 0 0;
  border-radius: 50%;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

<div
  className=" ListItemWrapper item__list__item  c0"
>
  <div
    className="ItemAvatar avatar__outer__wrapper c1"
  >
    <div
      className="avatar__wrapper online "
    />
  </div>
  <div
    className="ItemDetail item__details c2"
  />
</div>
`;
